elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.pagination;
+
+import com.yahoo.elide.core.exceptions.InvalidValueException;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Encapsulates the pagination strategy.
+ */
+
+@AllArgsConstructor
+@ToString
+public class Pagination {
+
+    /**
+     * Denotes the internal field names for paging.
+     */
+    public enum PaginationKey { page, pageSize }
+
+    public static final int DEFAULT_PAGE_SIZE = 500;
+    public static final int MAX_PAGE_SIZE = 10000;
+
+    private static final Pagination DEFAULT_PAGINATION = new Pagination(0, 0);
+
+    public static final Map<String, PaginationKey> PAGE_KEYS = new HashMap<>();
+    static {
+        PAGE_KEYS.put("page[size]", PaginationKey.pageSize);
```

Comment:
I think you need to differentiate in the parsing - and store the correct value.  Right now everything maps to page and pageSize.  This class should ultimately store the offset and limit (because that is what is passed to hibernate).  It alternatively could store both (offset & limit, page & pageSize), but that might just complicate things.